x-pack/plugins/watcher/public/models/action/index.action.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { get } from 'lodash';
+import { BaseAction } from './base_action';
+import { i18n } from '@kbn/i18n';
+
+const requiredFields = ['host', 'port'];
+const optionalFields = [
+  'scheme',
+  'path',
+  'method',
+  'headers',
+  'params',
+  'auth',
+  'body',
+  'proxy',
+  'connection_timeout',
+  'read_timeout',
+  'url'
+];
+
+const allFields = [...requiredFields, ...optionalFields];
+
+export class IndexAction extends BaseAction {
+  constructor(props = {}) {
+    super(props);
+
+    allFields.forEach((field) => {
+      this[field] = get(props, field);
+    });
+  }
+
+  get upstreamJson() {
```

Comment:
Yes you are right this could be optimized but as Watcher will be rewritten I just wanted to prevent future bugs in the meantime because this file was missing.